apps/passport-client/src/dispatch.ts

```diff
@@ -659,61 +681,54 @@ async function sync(state: AppState, update: ZuUpdate) {
       console.log(`[SYNC] failed to load issued PCDs, skipping this step`, e);
     }
 
-    update({
-      loadingIssuedPCDs: false,
+    return {
       loadedIssuedPCDs: true,
+      loadingIssuedPCDs: false,
       pcds: state.pcds,
       subscriptions: state.subscriptions
-    });
-    return;
-  }
-
-  if (!state.loadedIssuedPCDs && state.loadingIssuedPCDs) {
-    return;
+    };
   }
 
-  // Generate an upload ID from the state of PCDs and subscriptions
+  // Generate an upload ID from the state of PCDs and subscriptions.
+  // Upload only if the ID is different, meaning changes to upload.
   const uploadId = await makeUploadId(state.pcds, state.subscriptions);
+  if (state.uploadedUploadId !== uploadId) {
+    // Uploading requires state.self be set, which should be set by now.  If
+    // it's not, wait to upload on another sync triggered when self changes.
+    if (!state.self) {
+      console.error("[SYNC] no user available to upload");
+      return undefined;
+    }
 
-  // If it matches what we downloaded or uploaded already, there's nothing
-  // to do
-  if (
-    state.uploadedUploadId === uploadId ||
-    state.uploadingUploadId === uploadId
-  ) {
-    console.log("[SYNC] sync action: no-op");
-    return;
+    console.log("[SYNC] sync action: upload");
+    // TODO(artwyman): Add serverStorageRevision input here, but only after
+    // we're able to respond to a conflict by downloading.
```

Comment:
did you intend to get to that in this PR?

Reply:
No, this is for later, with a reminder of what's necessary to get it finished.  I think conflict resolution is going to be a Phase 2 or later thing, and we're still in Phase 1.  We could in theory use this just to detect conflicts, then either block the upload, or do an upload without knownRevision and clobber anyway, but I don't have useful progress on that in this diff.